crates/scroll/alloy/evm/src/block/mod.rs

```diff
@@ -95,13 +107,16 @@ where
             self.spec.is_spurious_dragon_active_at_block(self.evm.block().number);
         self.evm.db_mut().set_state_clear_flag(state_clear_flag);
 
-        // load the l1 gas oracle contract in cache
+        // load the l1 gas oracle contract in cache.
         let _ = self
             .evm
             .db_mut()
             .load_cache_account(L1_GAS_PRICE_ORACLE_ADDRESS)
             .map_err(BlockExecutionError::other)?;
 
+        // apply eip-2935.
+        self.system_caller.apply_blockhashes_contract_call(self.ctx.parent_hash, &mut self.evm)?;
+
```

Comment:
I think we need to add this to the payload builder, and maybe other places too, see instances of `builder.apply_pre_execution_changes` in upstream.